research/gam/run_configs.txt
This file contains the commands used to replicate the experiments in our paper. 

These configurations were obtained based on the best validation accuracy. Note that these results were obtained using Python 3.7 with Tensorflow 1.14.0. Running with other Tensorflow versions may lead to slight differences in results due to randomness. 

We recommend running on a GPU. With CUDA, this can be done by prepending "CUDA_VISIBLE_DEVICES=<your-gpu-number>" in front of each of the run commands below.

To visualize the results in Tensorboard, use the following command, adjusting the dataset name accordingly:
tensorboard --logdir=outputs/summaries/cora

--------------- Cora ----------------
MLP_128
    Baseline:
        First co-train iteration (i.e. iteration 0) of any of the runs below.
    NGM:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=cora  --model_cls=mlp --model_agr=mlp --hidden_cls=128 -hidden_agr=128 --batch_size_cls=128 --batch_size_agr=128  --num_pairs_reg=128 --first_iter_original=True --num_samples_to_label=200  --max_num_iter_cotrain=1 --experiment_suffix='-NGM'   --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --always_agree=True --seed=1234 --use_graph=True --penalize_neg_agr=False
    VAT:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=cora  --model_cls=mlp --model_agr=mlp --hidden_cls=128 -hidden_agr=128 --batch_size_cls=128 --batch_size_agr=128  --num_pairs_reg=128 --first_iter_original=True --num_samples_to_label=200  --max_num_iter_cotrain=1 --experiment_suffix='-NGM'   --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --always_agree=True --seed=1234 --use_graph=True --penalize_neg_agr=False --reg_weight_ll=0 --reg_weight_lu=0 --reg_weight_uu=0 --reg_weight_vat=1.0 --use_ent_min=False
    VATENT:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=cora  --model_cls=mlp --model_agr=mlp --hidden_cls=128 -hidden_agr=128 --batch_size_cls=128 --batch_size_agr=128  --num_pairs_reg=128 --first_iter_original=True --num_samples_to_label=200  --max_num_iter_cotrain=1 --experiment_suffix='-NGM'   --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --always_agree=True --seed=1234 --use_graph=True --penalize_neg_agr=False --reg_weight_ll=0 --reg_weight_lu=0 --reg_weight_uu=0 --reg_weight_vat=1.0 --use_ent_min=True
    GAM:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=cora  --model_cls=mlp --model_agr=mlp --hidden_cls=128 -hidden_agr=128 --batch_size_cls=128 --batch_size_agr=128 --num_pairs_reg=128 --first_iter_original=True --add_negative_edges_agr=True --aggregation_agr_inputs=dist --num_samples_to_label=200 --max_num_iter_cotrain=100 --first_iter_original=True --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --reg_weight_lu=1 --reg_weight_uu=0.5 --always_agree=False --seed=111 --keep_label_proportions=True --use_graph=True --penalize_neg_agr=False --min_confidence_new_label=0.9 --experiment_suffix='-GAM'
    GAM*:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=cpra  --model_cls=mlp --model_agr=mlp --hidden_cls=128 -hidden_agr=128 --batch_size_cls=128 --batch_size_agr=128  --num_pairs_reg=128 --first_iter_original=True --add_negative_edges_agr=True  --aggregation_agr_inputs=dist  --num_samples_to_label=200  --max_num_iter_cotrain=1000 --experiment_suffix='-GAM_star'  --first_iter_original=True --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --reg_weight_lu=100 --reg_weight_uu=50 --always_agree=False --seed=1234 --keep_label_proportions=True --use_graph=False --penalize_neg_agr=True

MLP_32_32_32_32:
    Baseline:
        First co-train iteration (i.e. iteration 0) of any of the runs below.
    NGM:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=cora  --model_cls=mlp --model_agr=mlp --hidden_cls=32_32_32_32 -hidden_agr=32_32_32_32 --batch_size_cls=128 --batch_size_agr=128  --num_pairs_reg=512 --first_iter_original=True --num_samples_to_label=200  --max_num_iter_cotrain=1 --experiment_suffix='-NGM'   --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --always_agree=True --seed=1234 --use_graph=True --penalize_neg_agr=False
    GAM:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=cora --model_cls=mlp --model_agr=mlp --hidden_cls=32_32_32_32 -hidden_agr=32_32_32_32 --batch_size_cls=128 --batch_size_agr=128 --num_pairs_reg=128 --first_iter_original=True --add_negative_edges_agr=True --aggregation_agr_inputs=dist --num_samples_to_label=200 --max_num_iter_cotrain=1000 --experiment_suffix='-GAM' --first_iter_original=True --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --reg_weight_lu=1000 --reg_weight_uu=500 --always_agree=False --seed=1234 --keep_label_proportions=True --use_graph=True --penalize_neg_agr=False
    GAM*:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=cora  --model_cls=mlp --model_agr=mlp --hidden_cls=32_32_32_32 -hidden_agr=32_32_32_32 --batch_size_cls=128 --batch_size_agr=128  --num_pairs_reg=128 --first_iter_original=True --add_negative_edges_agr=True  --aggregation_agr_inputs=dist  --num_samples_to_label=200  --max_num_iter_cotrain=1000 --experiment_suffix='-GAM_star'  --first_iter_original=True --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --reg_weight_lu=10 --reg_weight_uu=5 --always_agree=False --seed=1234 --keep_label_proportions=True --use_graph=False --penalize_neg_agr=True
    VAT:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=cora  --model_cls=mlp --model_agr=mlp --hidden_cls=32_32_32_32 -hidden_agr=32_32_32_32 --batch_size_cls=128 --batch_size_agr=128  --num_pairs_reg=128 --first_iter_original=True --num_samples_to_label=200  --max_num_iter_cotrain=1 --experiment_suffix='-NGM'   --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --always_agree=True --seed=1234 --use_graph=True --penalize_neg_agr=False --reg_weight_ll=0 --reg_weight_lu=0 --reg_weight_uu=0 --reg_weight_vat=1.0 --use_ent_min=False
    VATENT:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=cora  --model_cls=mlp --model_agr=mlp --hidden_cls=32_32_32_32 -hidden_agr=32_32_32_32 --batch_size_cls=128 --batch_size_agr=128  --num_pairs_reg=128 --first_iter_original=True --num_samples_to_label=200  --max_num_iter_cotrain=1 --experiment_suffix='-VATENT'   --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --always_agree=True --seed=1234 --use_graph=True --penalize_neg_agr=False --reg_weight_ll=0 --reg_weight_lu=0 --reg_weight_uu=0 --reg_weight_vat=1.0 --use_ent_min=True

GCN_128:
    Baseline:
        First co-train iteration (i.e. iteration 0) of any of the runs below.
    NGM:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=cora --model_cls=gcn --model_agr=mlp --hidden_cls=128 --hidden_agr=128 --batch_size_cls=128 --batch_size_agr=128 --num_pairs_reg=512 --first_iter_original=True --num_samples_to_label=200 --max_num_iter_cotrain=1 --experiment_suffix='-NGM' --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --always_agree=True --seed=1234 --use_graph=True --penalize_neg_agr=False
    GAM:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=cora  --model_cls=gcn --model_agr=mlp --hidden_cls=128 --hidden_agr=128 --num_pairs_reg=512 --first_iter_original=True --add_negative_edges_agr=True --aggregation_agr_inputs=dist --num_samples_to_label=200  --max_num_iter_cotrain=1000 --first_iter_original=True --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --experiment_suffix='-replaceLwithPred-noLabelProp' --num_iter_after_best_val_cls=2000 --row_normalize=False --reg_weight_lu=1 --reg_weight_uu=0.5 --always_agree=False --seed=111 --keep_label_proportions=False --use_graph=True --penalize_neg_agr=False --experiment_suffix='-GAM'
    GAM*:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=cora  --model_cls=gcn --model_agr=mlp --hidden_cls=128 --hidden_agr=128 --batch_size_cls=128 --batch_size_agr=128  --num_pairs_reg=512 --first_iter_original=True --add_negative_edges_agr=True  --aggregation_agr_inputs=dist  --num_samples_to_label=200  --max_num_iter_cotrain=1000 --experiment_suffix='-GAM_star'  --first_iter_original=True --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --reg_weight_lu=10.0 --reg_weight_uu=5 --always_agree=False --seed=1234  --keep_label_proportions=True --use_graph=False
    VAT:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=cora  --model_cls=gcn --model_agr=mlp --hidden_cls=128 --hidden_agr=128 --batch_size_cls=128 --batch_size_agr=128  --num_pairs_reg=128 --first_iter_original=True --num_samples_to_label=200  --max_num_iter_cotrain=1 --experiment_suffix='-VAT'   --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --always_agree=True --seed=1234 --use_graph=True --penalize_neg_agr=False --reg_weight_ll=0 --reg_weight_lu=0 --reg_weight_uu=0 --reg_weight_vat=1.0 --use_ent_min=False
    VATENT:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=cora  --model_cls=gcn --model_agr=mlp --hidden_cls=128 --hidden_agr=128 --batch_size_cls=128 --batch_size_agr=128  --num_pairs_reg=128 --first_iter_original=True --num_samples_to_label=200  --max_num_iter_cotrain=1 --experiment_suffix='-VATENT'  --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --always_agree=True --seed=1234 --use_graph=True --penalize_neg_agr=False --reg_weight_ll=0 --reg_weight_lu=0 --reg_weight_uu=0 --reg_weight_vat=1.0 --use_ent_min=True --first_iter_original=False

--------------- Citeseer ----------------
MLP_128:
    Baseline:
        First co-train iteration (i.e. iteration 0) of any of the runs below.
    NGM:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=citeseer  --model_cls=mlp --model_agr=mlp --hidden_cls=128 -hidden_agr=128 --batch_size_cls=128 --batch_size_agr=128  --num_pairs_reg=128 --first_iter_original=True --num_samples_to_label=200  --max_num_iter_cotrain=1 --experiment_suffix='-NGM'   --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --always_agree=True --seed=1234 --use_graph=True --penalize_neg_agr=False
    GAM:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=citeseer  --model_cls=mlp --model_agr=mlp --hidden_cls=128 -hidden_agr=128 --batch_size_cls=128 --batch_size_agr=128  --num_pairs_reg=128 --first_iter_original=True --add_negative_edges_agr=True  --aggregation_agr_inputs=dist  --num_samples_to_label=50  --max_num_iter_cotrain=100 --first_iter_original=True --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --reg_weight_lu=0.1 --reg_weight_uu=0.05 --always_agree=False --seed=333 --keep_label_proportions=True --use_graph=True --penalize_neg_agr=False --min_confidence_new_label=0.9 --experiment_suffix='-GAM'
    GAM*:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=citeseer  --model_cls=mlp --model_agr=mlp --hidden_cls=128 -hidden_agr=128 --batch_size_cls=128 --batch_size_agr=128  --num_pairs_reg=512 --first_iter_original=True --add_negative_edges_agr=True  --aggregation_agr_inputs=dist  --num_samples_to_label=200  --max_num_iter_cotrain=1000 --experiment_suffix='-GAM_star'   --first_iter_original=True --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --reg_weight_lu=10 --reg_weight_uu=5 --always_agree=False --seed=1234
    VAT:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=citeseer  --model_cls=mlp --model_agr=mlp --hidden_cls=128 -hidden_agr=128 --batch_size_cls=128 --batch_size_agr=128  --num_pairs_reg=128 --first_iter_original=True --num_samples_to_label=200  --max_num_iter_cotrain=1 --experiment_suffix='-VATENT'   --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --always_agree=True --seed=1234 --use_graph=True --penalize_neg_agr=False --reg_weight_ll=0 --reg_weight_lu=0 --reg_weight_uu=0 --reg_weight_vat=1.0 --use_ent_min=False
    VATENT:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=citeseer  --model_cls=mlp --model_agr=mlp --hidden_cls=128 -hidden_agr=128 --batch_size_cls=128 --batch_size_agr=128  --num_pairs_reg=128 --first_iter_original=True --num_samples_to_label=200  --max_num_iter_cotrain=1 --experiment_suffix='-VAT   --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --always_agree=True --seed=1234 --use_graph=True --penalize_neg_agr=False --reg_weight_ll=0 --reg_weight_lu=0 --reg_weight_uu=0 --reg_weight_vat=1.0 --use_ent_min=True

MLP_32_32_32_32:
    Baseline:
        First co-train iteration (i.e. iteration 0) of any of the runs below.
    NGM:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=citeseer  --model_cls=mlp --model_agr=mlp --hidden_cls=32_32_32_32 -hidden_agr=32_32_32_32 --batch_size_cls=128 --batch_size_agr=128  --num_pairs_reg=512 --first_iter_original=True --num_samples_to_label=200  --max_num_iter_cotrain=1 --experiment_suffix='-NGM'   --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --always_agree=True --seed=1234 --use_graph=True --penalize_neg_agr=False
    GAM:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=citeseer  --model_cls=mlp --model_agr=mlp --hidden_cls=32_32_32_32 -hidden_agr=32_32_32_32 --batch_size_cls=128 --batch_size_agr=128  --num_pairs_reg=512 --first_iter_original=True --add_negative_edges_agr=True  --aggregation_agr_inputs=dist  --num_samples_to_label=200  --max_num_iter_cotrain=1000 --experiment_suffix='-GAM'  --first_iter_original=True --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --reg_weight_lu=10 --reg_weight_uu=5 --always_agree=False --seed=1234 --keep_label_proportions=True --use_graph=True --penalize_neg_agr=False --experiment_suffix='-GAM'
    GAM*:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=citeseer  --model_cls=mlp --model_agr=mlp --hidden_cls=32_32_32_32 -hidden_agr=32_32_32_32 --batch_size_cls=128 --batch_size_agr=128  --num_pairs_reg=512 --first_iter_original=True --add_negative_edges_agr=True  --aggregation_agr_inputs=dist  --num_samples_to_label=200  --max_num_iter_cotrain=1000 --experiment_suffix='-GAM_star'  --first_iter_original=True --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --reg_weight_lu=1.0 --reg_weight_uu=0.5 --always_agree=False --seed=1234 --keep_label_proportions=True --use_graph=False --penalize_neg_agr=False
    VAT:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=citeseer  --model_cls=mlp --model_agr=mlp --hidden_cls=32_32_32_32 -hidden_agr=32_32_32_32 --batch_size_cls=128 --batch_size_agr=128  --num_pairs_reg=128 --first_iter_original=True --num_samples_to_label=200  --max_num_iter_cotrain=1 --experiment_suffix='-VAT'   --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --always_agree=True --seed=1234 --use_graph=True --penalize_neg_agr=False --reg_weight_ll=0 --reg_weight_lu=0 --reg_weight_uu=0 --reg_weight_vat=1.0 --use_ent_min=False
    VATENT:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=citeseer  --model_cls=mlp --model_agr=mlp --hidden_cls=32_32_32_32 -hidden_agr=32_32_32_32 --batch_size_cls=128 --batch_size_agr=128  --num_pairs_reg=128 --first_iter_original=True --num_samples_to_label=200  --max_num_iter_cotrain=1 --experiment_suffix='-VATENT'   --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --always_agree=True --seed=1234 --use_graph=True --penalize_neg_agr=False --reg_weight_ll=0 --reg_weight_lu=0 --reg_weight_uu=0 --reg_weight_vat=1.0 --use_ent_min=True

GCN_128:
    Baseline:
        First co-train iteration (i.e. iteration 0) of any of the runs below.
    NGM:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=citeseer --model_cls=gcn --model_agr=mlp --hidden_cls=128 --hidden_agr=128 --batch_size_cls=128 --batch_size_agr=128 --num_pairs_reg=512 --first_iter_original=True --num_samples_to_label=200 --max_num_iter_cotrain=1 --experiment_suffix='-NGM' --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --always_agree=True --seed=1234 --use_graph=True --penalize_neg_agr=False
    GAM:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=citeseer --model_cls=gcn --model_agr=mlp --hidden_cls=128 --hidden_agr=128 --num_pairs_reg=512 --first_iter_original=True --add_negative_edges_agr=True --aggregation_agr_inputs=dist --num_samples_to_label=200 --max_num_iter_cotrain=50 --first_iter_original=True --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=True --reg_weight_lu=0.01 --reg_weight_uu=0.005 --always_agree=False --seed=111 --keep_label_proportions=True --use_graph=True --penalize_neg_agr=False --experiment_suffix='-GAM'
    GAM*:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=citeseer  --model_cls=gcn --model_agr=mlp --hidden_cls=128 --hidden_agr=128 --batch_size_cls=128 --batch_size_agr=128  --num_pairs_reg=512 --first_iter_original=True --add_negative_edges_agr=True  --aggregation_agr_inputs=dist  --num_samples_to_label=200  --max_num_iter_cotrain=1000 --experiment_suffix='-GAM_star'  --first_iter_original=True --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --reg_weight_lu=10.0 --reg_weight_uu=5 --always_agree=False --seed=1234  --keep_label_proportions=True --use_graph=False
    VAT:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=citeseer  --model_cls=gcn --model_agr=mlp --hidden_cls=128 --hidden_agr=128 --batch_size_cls=128 --batch_size_agr=128  --num_pairs_reg=128 --first_iter_original=True --num_samples_to_label=200  --max_num_iter_cotrain=1 --experiment_suffix='-VAT'   --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --always_agree=True --seed=1234 --use_graph=True --penalize_neg_agr=False --reg_weight_ll=0 --reg_weight_lu=0 --reg_weight_uu=0 --reg_weight_vat=1.0 --use_ent_min=False
    VATENT:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=citeseer  --model_cls=gcn --model_agr=mlp --hidden_cls=128 --hidden_agr=128 --batch_size_cls=128 --batch_size_agr=128  --num_pairs_reg=128 --first_iter_original=True --num_samples_to_label=200  --max_num_iter_cotrain=1 --experiment_suffix='-VATENT'   --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --always_agree=True --seed=1234 --use_graph=True --penalize_neg_agr=False --reg_weight_ll=0 --reg_weight_lu=0 --reg_weight_uu=0 --reg_weight_vat=1.0 --use_ent_min=True

--------------- Pubmed ----------------
MLP_128:
    Baseline:
        First co-train iteration (i.e. iteration 0) of any of the runs below.
    NGM:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=pubmed --model_cls=mlp --model_agr=mlp --hidden_cls=128 -hidden_agr=128 --batch_size_cls=128 --batch_size_agr=128 --num_pairs_reg=128 --first_iter_original=True --num_samples_to_label=200 --max_num_iter_cotrain=1 --experiment_suffix='-NGM' --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --always_agree=True --seed=1234 --use_graph=True --penalize_neg_agr=False
    GAM:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=pubmed --model_cls=mlp --model_agr=mlp --hidden_cls=128 -hidden_agr=128 --batch_size_cls=128 --batch_size_agr=128 --num_pairs_reg=128 --first_iter_original=True --add_negative_edges_agr=True --aggregation_agr_inputs=dist --num_samples_to_label=200 --max_num_iter_cotrain=1000 --use_l2_cls=False --first_iter_original=True --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --reg_weight_lu=0.1 --reg_weight_uu=0.05 --keep_label_proportions=False --use_graph=True --penalize_neg_agr=False --seed=111 --experiment_suffix='-GAM'
    GAM*:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=pubmed  --model_cls=mlp --model_agr=mlp --hidden_cls=128 -hidden_agr=128 --batch_size_cls=128 --batch_size_agr=128  --num_pairs_reg=512 --first_iter_original=True --add_negative_edges_agr=True  --aggregation_agr_inputs=dist  --num_samples_to_label=200  --max_num_iter_cotrain=1000  --experiment_suffix='-GAM_star'  --first_iter_original=True --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --reg_weight_lu=1000 --reg_weight_uu=500 --always_agree=False --seed=1234
    VAT:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=pubmed  --model_cls=mlp --model_agr=mlp --hidden_cls=128 -hidden_agr=128 --batch_size_cls=128 --batch_size_agr=128  --num_pairs_reg=128 --first_iter_original=True --num_samples_to_label=200  --max_num_iter_cotrain=1 --experiment_suffix='-VAT'   --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --always_agree=True --seed=1234 --use_graph=True --penalize_neg_agr=False --reg_weight_ll=0 --reg_weight_lu=0 --reg_weight_uu=0 --reg_weight_vat=1.0 --use_ent_min=False
    VATENT:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=pubmed  --model_cls=mlp --model_agr=mlp --hidden_cls=128 -hidden_agr=128 --batch_size_cls=128 --batch_size_agr=128  --num_pairs_reg=128 --first_iter_original=True --num_samples_to_label=200  --max_num_iter_cotrain=1 --experiment_suffix='-VATENT'   --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --always_agree=True --seed=1234 --use_graph=True --penalize_neg_agr=False --reg_weight_ll=0 --reg_weight_lu=0 --reg_weight_uu=0 --reg_weight_vat=1.0 --use_ent_min=True

MLP_32_32_32_32:
    Baseline:
        First co-train iteration (i.e. iteration 0) of any of the runs below.
    NGM:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=pubmed --model_cls=mlp --model_agr=mlp --hidden_cls=32_32_32_32 -hidden_agr=32_32_32_32 --batch_size_cls=128 --batch_size_agr=128 --num_pairs_reg=512 --first_iter_original=True --num_samples_to_label=200 --max_num_iter_cotrain=1 --experiment_suffix='-NGM' --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --always_agree=True --seed=1234 --use_graph=True --penalize_neg_agr=False
    GAM:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=pubmed  --model_cls=mlp --model_agr=mlp --hidden_cls=32_32_32_32 -hidden_agr=32_32_32_32 --batch_size_cls=128 --batch_size_agr=128  --num_pairs_reg=512 --first_iter_original=True --add_negative_edges_agr=True  --aggregation_agr_inputs=dist  --num_samples_to_label=200  --max_num_iter_cotrain=1000 --experiment_suffix='-GAM'  --first_iter_original=True --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --reg_weight_lu=1000 --reg_weight_uu=500 --always_agree=False --seed=1234 --keep_label_proportions=True --use_graph=True --penalize_neg_agr=False --reg_weight_vat=0
    GAM*:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=pubmed  --model_cls=mlp --model_agr=mlp --hidden_cls=32_32_32_32 -hidden_agr=32_32_32_32 --batch_size_cls=128 --batch_size_agr=128  --num_pairs_reg=512 --first_iter_original=True --add_negative_edges_agr=True  --aggregation_agr_inputs=dist  --num_samples_to_label=200  --max_num_iter_cotrain=1000 --experiment_suffix='-GAM_star'  --first_iter_original=True --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --reg_weight_lu=1.0 --reg_weight_uu=0.5 --always_agree=False --seed=1234
    VAT:
        python3.7 -m gam.experiments.run_train_gam_graph --data_path=<path_to_data> --dataset_name=pubmed  --model_cls=mlp --model_agr=mlp --hidden_cls=32_32_32_32 -hidden_agr=32_32_32_32 --batch_size_cls=128 --batch_size_agr=128  --num_pairs_reg=128 --first_iter_original=True --num_samples_to_label=200  --max_num_iter_cotrain=1 --experiment_suffix='-VAT'   --learning_rate_cls=1e-3 --num_iter_after_best_val_agr=2000 --num_iter_after_best_val_cls=2000 --row_normalize=False --always_agree=True --seed=1234 --use_graph=True --penalize_neg_agr=False --reg_weight_ll=0 --reg_weight_lu=0 --reg_weight_uu=0 --reg_weight_vat=1.0 --use_ent_min=False
    VATENT:

research/kg_hyp_emb/config.py
# Copyright 2020 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     https://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
"""Default configuration parameters."""

from __future__ import absolute_import
from __future__ import division
from __future__ import print_function

CONFIG = {
    'string': {
        'dataset': ('Dataset', 'WN18RR'),
        'model': ('Model', 'RotE'),
        'data_dir': ('Path to data directory', 'data/'),
        'save_dir': ('Path to logs directory', 'logs/'),
        'loss_fn': ('Loss function to use', 'SigmoidCrossEntropy'),
        'initializer': ('Which initializer to use', 'GlorotNormal'),
        'regularizer': ('Regularizer', 'N3'),
        'optimizer': ('Optimizer', 'Adam'),
        'bias': ('Bias term', 'learn'),
        'dtype': ('Precision to use', 'float32'),
    },
    'float': {
        'lr': ('Learning rate', 1e-3),
        'lr_decay': ('Learning rate decay', 0.96),
        'min_lr': ('Minimum learning rate decay', 1e-5),
        'gamma': ('Margin for distance-based losses', 0),
        'entity_reg': ('Regularization weight for entity embeddings', 0),
        'rel_reg': ('Regularization weight for relation embeddings', 0),
    },
    'integer': {
        'patience': ('Number of validation steps before early stopping', 20),
        'valid': ('Number of epochs before computing validation metrics', 5),
        'checkpoint': ('Number of epochs before checkpointing the model', 5),
        'max_epochs': ('Maximum number of epochs to train for', 400),
        'rank': ('Embeddings dimension', 500),
        'batch_size': ('Batch size', 500),
        'neg_sample_size':
            ('Negative sample size, -1 to use loss without negative sampling',
             50),
    },
    'boolean': {
        'train_c': ('Whether to train the hyperbolic curvature or not', True),
        'debug': ('If debug is true, only use 1000 examples for'
                  ' debugging purposes', False),
        'save_logs':
            ('Whether to save the training logs or print to stdout', True),
        'save_model': ('Whether to save the model weights', False)
    }
}
